cluster/engine.go

```diff
@@ -434,7 +459,7 @@ func (e *Engine) refreshLoop() {
 			failedAttempts++
```

Comment:
You are right. I merge this logic and update `e.opts.RefreshRetry` to `e.opts.FailureRetry`.